src/game/level.c

```diff
@@ -710,27 +710,29 @@ static void Level_LoadAnimatedTextures(MYFILE *fp)
 
         // Level data is tied to the original logic in Output_AnimateTextures
         // and hence stores one less than the actual count here.
-        range->num_textures = File_ReadS16(fp);
+        range->num_textures = VFile_ReadS16(file);
         range->num_textures++;
 
         range->textures = GameBuf_Alloc(
             sizeof(int16_t) * range->num_textures,
             GBUF_ANIMATING_TEXTURE_RANGES);
-        File_ReadItems(
-            fp, range->textures, sizeof(int16_t), range->num_textures);
+        VFile_Read(
+            file, range->textures, sizeof(int16_t) * range->num_textures);
     }
 
 cleanup:
     // Ensure to read everything intended by the level compiler, even if it
     // does not wholly contain accurate texture data.
-    File_Seek(fp, MAX(end_position, File_Pos(fp)), SEEK_SET);
+    if (end_position - VFile_GetPos(file) > 0) {
```

Comment:
This should be something like the following I think. The EGYPT.PHD file in #1346 crashes.
```c
const int32_t skip_length = end_position - VFile_GetPos(file);
if (skip_length > 0) {
    VFile_Skip(file, skip_length);
}
```

Or we implement `VFile_Seek` and use the previous `MAX` approach.

Reply:
Ok I added in your suggestion thanks!